### PR TITLE
Updated setup.md

### PR DESCRIPTION
**Pull request by kakshay21**
Added installation guide for fedora
update the markdown error in the same file
Add Installation Guide for different Platforms #866
@deshraj @TroJan

### CODE REVIEW

**Comment by TroJan** [comment]
@kakshay21 : Can you close one of the PR(the one which is not relevant). Also please categorize according to OS like first write instructions for Ubuntu and then Fedora instead of categorizing each install step per OS. This way person setting up EvalAI for the first time on his system can directly go to heading of OS and just follow the instructions.

**Comment by kakshay21** [comment]
okay @TroJan

File: docs/setup.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ## How to setup
 
-This section guides first-time contributor through installing the EvalAI development environment on Ubuntu(recommended 14.04)
+This section guides first-time contributor through installing the EvalAI development environment on <h4>Ubuntu(recommended 14.04)</h4>
```
Comment (deshraj) [inline docs/setup.md]: Please replace this line with 

```
EvalAI can run on Linux, Cloud, Windows, and macOS platforms. Use the following list to choose the best installation path for you. The links under *Platform* take you straight to the installation instructions for that platform.

- Ubuntu (hyperlink this part to the Ubuntu installation heading)
- Fedora (hyperlink this part to the fedora installation heading)
.....so on.
```

File: docs/setup.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
```
Comment (deshraj) [inline docs/setup.md]: Add 
```
## Ubuntu Installation Instructions
```

File: docs/setup.md [hunk 3]
```diff
@@ -102,6 +102,61 @@ python manage.py runserver --settings=settings.dev
 gulp dev:runserver
 ```
 
+And this section guides first-time contributor through installing the EvalAI development environment on <h4>Fedora(recommended 23+)</h4>
```
Comment (deshraj) [inline docs/setup.md]: Drop this line and replace with 

```
## Fedora Installation Instructions
```

**Comment by deshraj** [comment]
@kakshay21 Please keep us posted when you are done with the changes.

**Comment by kakshay21** [comment]
Oh..  It's done sir @deshraj

File: docs/setup.md [hunk 4]
```diff
@@ -1,8 +1,9 @@
-## How to setup
```
Comment (deshraj) [inline docs/setup.md]: Please don't delete this one.

**Comment by deshraj** [comment]
@kakshay21 most of the things are good except the links. They are not rendered correctly. Can you check https://github.com/kakshay21/EvalAI/blob/94fae400246281e31cbc1cdf5a64655bd0c07f03/docs/setup.md and make changes for the links?

**Comment by kakshay21** [comment]
Were you talking about links like this```https://127.0.0.1:8000```?
This one is fixed.
@TroJan @deshraj

**Comment by kakshay21** [comment]
Anything else, sir?
@deshraj

File: docs/setup.md [hunk 5]
```diff
@@ -86,24 +88,134 @@ npm install
...
+* To run backend development server at [http://127.0.0.1:8000](http://127.0.0.1:8000)
 
 ```
 # activate virtual environment if not activated
 source venv/bin/activate
 python manage.py runserver --settings=settings.dev
 ```
 
-* To run frontend development server for at `http://127.0.0.1:8888`[http://127.0.0.1:8888]
+* To run frontend development server for at [http://127.0.0.1:8000](http://127.0.0.1:8000)
```
Comment (deshraj) [inline docs/setup.md]: The port has to be `8888` and not `8000`

File: docs/setup.md [hunk 6]
```diff
@@ -86,24 +88,134 @@ npm install
...
+
+* To run backend development server at [http://127.0.0.1:8000](http://127.0.0.1:8000)
+
+```
+# activate virtual environment if not activated
+source venv/bin/activate
+python manage.py runserver --settings=settings.dev
+```
+
+* To run frontend development server for at [http://127.0.0.1:8000](http://127.0.0.1:8000)
```
Comment (deshraj) [inline docs/setup.md]: Please change port to `8888` from `8000`

Comment (kakshay21) [inline docs/setup.md]: my bad! sorry

**Comment by deshraj** [comment]
Looks good now. Thanks @kakshay21 for setting this up. :)

**Pull request merged**